File: DAL/Models/Apartment.cs
using DAL.Models;

namespace Utilities.Models
{
    public class Apartment
    {
        public int Id { get; set; }

        public string? RelatedFamily { get; set; }

        public int RegisteredCountPeople { get; set; }

        public string? ReceiptCode { get; set; }

        public virtual MeterDocument? MeterDocument { get; set; }

        public int AddressId { get; set; }

        public virtual Address? Address { get; set; }

        public virtual IEnumerable<Meter> Meters { get; set; }

        public Apartment()
        {
            Meters = new List<Meter>();
        }
    }
}

File: Utilities/ClientApp/src/app/Models/Apartment.ts
import { Meter } from "./Meter";
import { MeterDocument } from "./MeterDocument";

export interface Apartment {
    id: number;
    relatedFamily: string;
    registeredCountPeople: number;
    apartmentNumber: string;
    houseNumber: string;
    buildingNumber: number;
    entranceNumber: number;
    receiptCode?: string;
    meterDocument: MeterDocument;
    meters: Array<Meter>;
}